### ISSUE TITLE
Page header: Some components have no visible focus and missing labels

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3458]
When you tab to some header components they don't show where the focus is. That will make it harder for keyboard only users (like motor impaired people) to navigate.
That fails [2.4.7 Focus Visible](https://www.w3.org/TR/WCAG21/#focus-visible).

* [ ] The very first link on the page is a skip link which skips to the main content. But as that link is always visually hidden it will only benefit screen reader users and not other types of keyboard only users.
Best either show the link at least when it has focus, or always show it.

* [ ] The link around the logo and wiki title has no visible focus indicator.

* [ ] The search button in the search box has no visible focus indicator either. You might not realise this if you use Chrome as Chrome has a default button outline which DokuWiki doesn't reset. Try it in Firefox to see.

* [ ] The search input only has a placeholder. Some screen readers don't read a `placeholder` attribute. When I tested with voice control (Dragon) and said "click search", it would click the button and not the search box. (It works via other means, like saying "click box" or via the general mouse grid fallback.) It should use a `label` instead (which could be visually hidden) or `aria-label` etc.
That fails [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value).

## Other non-fail accessibility issues

These don't fail WCAG and are minor issues.

* [ ] The tools dropdown that you see in mobile view has a `title` but no proper `label`. A `title` is generally well supported by screen readers and voice control tools. But you might want to use a proper label anyway (which could be visually hidden).
![](https://user-images.githubusercontent.com/108893/113484351-c756d680-949f-11eb-8ec0-bc5d65b41780.png)

* [ ] The link around the wiki title and logo doesn't describe its purpose well enough. (In other wikis with other titles that might be clearer.) It is unclear that the link goes to the homepage. It would be helpful if there was something that indicates the purpose of the link (like a `title` saying "Homepage").
